### ISSUE TITLE
How to add a resource?

### ISSUE THREAD

**Issue by j-a-n** [aspectron/winres-edit#1]
Hello,

thanks for this useful crate!

I have already managed to replace resources in a binary.
Now I am trying to add mutiple icons from an icon file to a binary.
I can replace existing ICON resources, but I have trouble adding new ones.
To create a `Resource` I have to pass `module_handle` to `new`.
But `module_handle` is declared private in `Resources`.

After patching winres-edit and setting `module_handle` to public the following code adds a new icon to the binary:

```rust
let mut resources = Resources::new(&Path::new("myfile.exe"));
resources.open().expect("Unable to open resource file for updates");
let res = Resource::new(
	resource_type::ICON.into(),
	Id::Integer(9).into(),
	1033,
	icon_data,
	resources.module_handle.clone()
);
res.update().unwrap();
```

Is there a possibility that I have overlooked?

**Comment by aspect** [comment]
Looks like something is not working.  I'll try to find time to check tonight. The application I've developed this for (https://github.com/aspectron/cargo-nw/) does updates only, so it looks like I wasn't testing insertion much (but I know I was able to add icons at some point).

**Comment by aspect** [comment]
I've tested resource creation using the method you have described and it is working.

I've published `0.2.0` that slightly simplifies invoking `Resources::new()` by accepting the resource manager as the first argument. 

So now you can call:
```rust
let resources = Resources::new(&Path::new("myfile.exe"));
resources.open()?;
let res = Resources::new(&resources, ...);
res.update()?;
resources.close()?;
```

While this simplifies the usage, this does not solve your problem.

1) make sure to call `resources.close()` after you are done for the resource manager to flush changes (it gets called automatically via a `Drop` trait, but that can result in suppression of errors, albeit Win32 errors are not very useful, but at least this can indicate that something is wrong).
2) in general, the underlying Win32 API is very capricious, to say the least.  There could be something that it rejects during resource creation...  I would recommend using the library to extract the existing icon and analyzing it to make sure that the arguments you are specifying are correct.

For example, `1033` is the language resource identifier, which may not exist in your target application.

You can use Resource Hacker http://www.angusj.com/resourcehacker/ to examine the executable.  I bet the issue is that the resource you are supplying somehow doesn't "fit".

Another thing to try could be loading an existing icon from the executable and replacing its data and id (name), then trying to update it (the new id (name) will result in the new resource being created).

**Comment by j-a-n** [comment]
Thanks for your quick response!

I must have expressed myself in a misleading way.
The code I posted IS working for me.
The only problem I had was the fact that I could not use `resources.module_handle` because it was private.

I will try version 0.2.0.

**Comment by aspect** [comment]
Ah, then this is addressed in `0.2.0`.  Let me know if this works for you.

**Comment by j-a-n** [comment]
Just updated to version 0.2.0 and everything works perfectly.
Thanks again!
